src/lib/io/sol-iio.c

```diff
@@ -157,6 +157,8 @@ craft_filename_path(char *path, size_t size, const char *base, ...)
     len = vsnprintf(path, size, base, args);
     va_end(args);
 
+    SOL_DBG("Crafted path len: %d Buffer size: %zu", len, size);
```

Comment:
fair enough for the failure, BUT  on success you're not printing it :-) That's not fair.

You can start path[] = "", so if it's failed either snprintf will null-terminate or it will be already null-terminated.